e2e/testutils/plugins/basic.go

```diff
@@ -9,6 +9,11 @@ import (
 	"time"
 )
 
+// set by the compile flags to get around content sha being the same
+var (
+	UNIQUEME string
+)
+
```

Comment:
This one doesn't seem to be used anywhere, so could lead to a linter starting to flag it.

Possibly we could avoid that through something silly like `_ = UNIQUEME`, but slightly curious what the issue was that you ran into; did this cause issues in tests where Go was caching things? Or was there something else?

Reply:
The e2e tests build a plugin and pushes that to a registry run inside docker (spun up by the tests). That said, for multiple tests doing `docker plugin create` and `docker plugin push` you would end up with sha collisions since the source code of the plugin binary built would be the same. The `UNIQUEME` variable is a way around this, so we can re-build the same plugin binary from the same source code generating a new sha hash. 

See these tests for reference https://github.com/docker/cli/blob/7c722c08d093c118ea727961be9aa0a23c83b733/e2e/global/cli_test.go#L143-L181

Reply:
Oh! And now curious; wouldn't that be something we could take advantage of? If we're building (and pushing) the exact same plugin multiple times; would we be able to use that so that we don't have to? 🤔 


Wondering now; ISTR we have something similar in moby/moby, where we need to have an image that's built and pushed in CI (but have something to verify if it needs to be built / built only once). I'd have to go look for that though 🙈

Reply:
Not necessarily since there is also a test that verifies if it _can_ push the image

https://github.com/docker/cli/blob/master/e2e/plugin/trust_test.go#L37-L46

Reply:
Yes, definitely fine to look at separately